compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -52,6 +60,11 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     else nullables.toList
   }
 
+  private def needsBoxing(tp: Type)(using Context): Boolean = tp != NoType && tp != defn.UnitType && tp.classSymbol.isPrimitiveValueClass
+    
+  private def boxIfCan(tp: Type)(using Context): Type =
+    assert(needsBoxing(tp))
+    defn.boxedType(tp)
```

Comment:
This is used only once, in a context where `needsBoxing(tp)` is closely known to be `true`. Consider inlining `defn.boxedType(tp)` directly at call site.